initial-ai-src/script.js

```diff
@@ -0,0 +1,587 @@
+class RuterDiscountCalculator {
+  constructor() {
+    this.maxDiscount = 40; // Maximum discount percentage
+
+    // Zone-based single ticket prices (NOK)
+    this.zonePrices = {
+      1: 39, // Zone 1 only
+      2: 55, // Zones 1-2
+      3: 73, // Zones 1-3
+      4: 91, // Zones 1-4
+      all: 109, // All zones
+    };
+
+    // Zone-based period ticket prices (NOK)
+    this.periodTickets = {
+      1: {
+        day: { price: 110, name: "24-hour ticket", days: 1 },
+        week: { price: 280, name: "7-day ticket", days: 7 },
+        month: { price: 790, name: "30-day ticket", days: 30 },
+        year: { price: 7900, name: "Annual ticket", days: 365 },
+      },
+      2: {
+        day: { price: 155, name: "24-hour ticket", days: 1 },
+        week: { price: 395, name: "7-day ticket", days: 7 },
+        month: { price: 1110, name: "30-day ticket", days: 30 },
+        year: { price: 11100, name: "Annual ticket", days: 365 },
+      },
+      3: {
+        day: { price: 205, name: "24-hour ticket", days: 1 },
+        week: { price: 520, name: "7-day ticket", days: 7 },
+        month: { price: 1470, name: "30-day ticket", days: 30 },
+        year: { price: 14700, name: "Annual ticket", days: 365 },
+      },
+      4: {
+        day: { price: 255, name: "24-hour ticket", days: 1 },
+        week: { price: 650, name: "7-day ticket", days: 7 },
+        month: { price: 1830, name: "30-day ticket", days: 30 },
+        year: { price: 18300, name: "Annual ticket", days: 365 },
+      },
+      all: {
+        day: { price: 305, name: "24-hour ticket", days: 1 },
+        week: { price: 780, name: "7-day ticket", days: 7 },
+        month: { price: 2190, name: "30-day ticket", days: 30 },
+        year: { price: 21900, name: "Annual ticket", days: 365 },
+      },
+    };
+
+    this.init();
```

Comment:
❓ Why separate `init` from the constructor? By defining both `init` and `constructor` and using them this way, you've effectively created two constructor functions. Note: a separate `init` would make sense if it was called in other ways in other places.